### PR TITLE
Upgrade flow-bin to 0.53.1

### PR DESCRIPTION
**Pull request by bvaughn**
This keeps Flow versions in-sync between GitHub and fbsource. I noticed a Flow incompatibility between these 2 versions when syncing React Native this morning.

Commits are atomic to simplify the review process.

Resources:
* [Flow 53 release blog](https://medium.com/flow-type/even-better-support-for-react-in-flow-25b0a3485627)
* [Flow React type definitions](https://github.com/facebook/flow/blob/v0.53.1/lib/react.js)
* [`flow-upgrade` utility](https://yarnpkg.com/en/package/flow-upgrade)
* [Flow React type reference](https://flow.org/en/docs/react/types/)

### CODE REVIEW

**Comment by bvaughn** [comment]
Whoops. Looks like I missed some lint errors. Will fix shortly.

**Comment by bvaughn** [comment]
Tagging everyone for review (b'c no one's the obvious owner of this sort of change)

**Comment by flarnie** [comment]
thumb_up

File: flow/react-native-host-hooks.js [hunk 1]
```diff
@@ -83,5 +83,5 @@ declare module 'UIManager' {
   ) : Promise<any>;
 }
 declare module 'View' {
-  declare var exports : typeof ReactComponent;
+  declare var exports : typeof React$Component;
```
Comment (gaearon) [inline flow/react-native-host-hooks.js]: Can we use React.Component here instead? I thought that's the recommended alias.

File: src/renderers/dom/fiber/ReactDOMFiberEntry.js [hunk 2]
```diff
@@ -650,7 +650,7 @@ function renderSubtreeIntoContainer(
 
 var ReactDOMFiber = {
   hydrate(
-    element: ReactElement<any>,
+    element: React$Element<any>,
```
Comment (gaearon) [inline src/renderers/dom/fiber/ReactDOMFiberEntry.js]: Ditto. (And actually this should take React.Node I think).

File: flow/react-native-host-hooks.js (hunk 1 above)
Comment (bvaughn) [reply flow/react-native-host-hooks.js]: Based on [this list of React types in Flow](https://github.com/facebook/flow/blob/v0.53.1/lib/react.js), `React$Component` seems a legit type. Also, commit 3a920c6 was me running the [`flow-upgrade` utility](https://yarnpkg.com/en/package/flow-upgrade) provided by Flow, and it also inserted a bunch of `React$Component` types so... it seems okay?

File: src/renderers/dom/fiber/ReactDOMFiberEntry.js (hunk 2 above)
Comment (bvaughn) [reply src/renderers/dom/fiber/ReactDOMFiberEntry.js]: To be honest, I'm still a little unclear on when we should be using one vs the other, but I think you're right that the [`Node` type](https://flow.org/en/docs/react/types/#toc-react-node) would work here.

**Comment by bvaughn** [comment]
Okay. I rebased and changed `React$Element` to `React$Node`.

File: flow/react-native-host-hooks.js (hunk 1 above)
Comment (gaearon) [reply flow/react-native-host-hooks.js]: It is a legit type but it's the internal name. I don't think it's meant to be used directly. The "friendly" alias is recommended for real use.

Comment (gaearon) [reply flow/react-native-host-hooks.js]: I might be wrong though. Maybe @calebmer can tell.

Comment (bvaughn) [reply flow/react-native-host-hooks.js]: Interesting. So I just took a closer look at [the Flow upgrade script](https://unpkg.com/flow-upgrade@1.0.3/dist/upgrades/0.53.0/ReactUtilityTypes/codemod.js) and saw this comment:

> // Update the type identifier to be `React.Element` or the
> // equivalent global.

Seems like that implies the `React$Component` type is a "global" and is safe to use? Which lines up with the usage of it that I see everywhere.

Comment (gaearon) [reply flow/react-native-host-hooks.js]: Yes, they're global and safe to use. I didn't mean to say they're not working. It just looks ugly and if there's a "proper" way to use it, I'd prefer that. But it's not important and not blocking this diff.

Comment (bvaughn) [reply flow/react-native-host-hooks.js]: Gotcha. I believe I misunderstood what you meant by "internal name". 😄 Thanks for clarifying.

Seeing as how we were already using these types (via the `ReactComponent` and `ReactElement` aliases we declared) I think it's save to move forward with this PR and follow up later based on feedback from @calebmer!

File: src/renderers/shared/fiber/ReactFiberScheduler.js [hunk 3]
```diff
@@ -111,7 +111,9 @@ if (__DEV__) {
     stopCommitLifeCyclesTimer,
   } = require('ReactDebugFiberPerf');
 
-  var warnAboutUpdateOnUnmounted = function(instance: ReactClass<any>) {
+  var warnAboutUpdateOnUnmounted = function(
+    instance: React$ComponentType<any>,
```
Comment (gaearon) [inline src/renderers/shared/fiber/ReactFiberScheduler.js]: Doesn't look right to me. `React$ComponentType` represents a React *type*, not instance. E.g. `MyClass` or `MyFunctionalComponent`.

The original typing was wrong too though.

File: src/renderers/shared/fiber/ReactFiberScheduler.js [hunk 4]
```diff
@@ -123,7 +125,7 @@ if (__DEV__) {
     );
   };
 
-  var warnAboutInvalidUpdates = function(instance: ReactClass<any>) {
+  var warnAboutInvalidUpdates = function(instance: React$ComponentType<any>) {
```
Comment (gaearon) [inline src/renderers/shared/fiber/ReactFiberScheduler.js]: Same.

File: src/renderers/shared/fiber/ReactFiberScheduler.js (hunk 3 above)
Comment (bvaughn) [reply src/renderers/shared/fiber/ReactFiberScheduler.js]: Thanks for catching this Dan. 👍

**Pull request merged**